From 8cfde092bdee3ce0c969f54d5edfe69180e2b4e2 Mon Sep 17 00:00:00 2001
From: Peter Marschall <peter@adpm.de>
Date: Thu, 2 Jun 2011 18:20:58 +0200
Subject: [PATCH 22/26] opensc-explorer: replace if..else with ?:

Use the easier to read & shorter expression
  path->type = (is_id) ? SC_PATH_TYPE_FILE_ID : SC_PATH_TYPE_PATH;
nstead of the longer, but equivalent if () .. else construction.

Signed-off-by: Peter Marschall <peter@adpm.de>
---
 src/tools/opensc-explorer.c |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/src/tools/opensc-explorer.c b/src/tools/opensc-explorer.c
index 5753afe..2b83614 100644
--- a/src/tools/opensc-explorer.c
+++ b/src/tools/opensc-explorer.c
@@ -301,10 +301,7 @@ static int arg_to_path(const char *arg, sc_path_t *path, int is_id)
 		if ((cbuf[0] == 0x3F && cbuf[1] == 0x00) || is_id) {
 			path->len = 2;
 			memcpy(path->value, cbuf, 2);
-			if (is_id)
-				path->type = SC_PATH_TYPE_FILE_ID;
-			else
-				path->type = SC_PATH_TYPE_PATH;
+			path->type = (is_id) ? SC_PATH_TYPE_FILE_ID : SC_PATH_TYPE_PATH;
 		} else {
 			*path = current_path;
 			if (path->type == SC_PATH_TYPE_DF_NAME)   {
-- 
1.7.5.3

